examples/run_CPMG.py

```diff
@@ -31,91 +35,79 @@
-    nPoints = nPoints,
-    nEchoes = config_dict["nEchoes"],
-    p90_us = config_dict["p90_us"],
-    repetition_us = config_dict["repetition_us"],
-    tau_us = config_dict["tau_us"],
-    SW_kHz = config_dict["SW_kHz"],
-    pad_start_us = pad_start,
-    pad_end_us = pad_end,
-    ret_data = None,
+data = generic(
+    ppg_list=[
+        ("marker", "start", 1),
+        ("phase_reset", 1),
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", config_dict["p90_us"], "ph_cyc", ph1_cyc),
+        ("delay", tau_us),
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", 2.0 * config_dict["p90_us"], "ph_cyc", ph2_cyc),
+        ("delay", config_dict["deadtime_us"]),
+        ("acquire", acq),
+        ("delay", pad_end_us),
+        ("delay", short_delay_us),  # matching the jumpto delay
+        ("marker", "echo_label", (config_dict["nEchoes"] - 1)),
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", 2.0 * config_dict["p90_us"], "ph_cyc", ph2_cyc),
+        ("delay", config_dict["deadtime_us"]),
+        ("acquire", acq),
+        ("delay", pad_end_us),
+        ("jumpto", "echo_label"),
+        ("delay", config_dict["repetition_us"]),
+        ("jumpto", "start"),
+    ],
+    nScans=config_dict["nScans"],
+    indirect_idx=0,
+    indirect_len=len(config_dict["nEchoes"]),
+    adcOffset=config_dict["adc_offset"],
+    carrierFreq_MHz=config_dict["carrierFreq_MHz"],
+    nPoints=nPoints,
+    SW_kHz=config_dict["SW_kHz"],
+    ret_data=None,
 )
 # }}}
 # {{{ chunk and save data
-if phase_cycling:
-    data.chunk("t", ["ph1", "echo", "t2"], [len(ph1_cyc), config_dict["nEchoes"], -1])
-    data.setaxis("ph1", ph1_cyc / 4)
-    data.setaxis("echo", r_[0 : config_dict["nEchoes"]])
-    if config_dict["nScans"] > 1:
-        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-    data.squeeze()
-    data.set_units("t2", "s")
-    fl.next("Raw - time")
-    fl.image(
-        data.C.mean("nScans"))
-    data.reorder("t2", first=False)
-    for_plot = data.C
-    for_plot.ft('t2',shift=True)
-    for_plot.ft(['ph1'], unitary = True)
-    fl.next('FTed data')
-    fl.image(for_plot.C.mean("nScans")
-    )
-else:
-    if config_dict["nScans"] > 1:
-        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-    data.rename("t","t2")
-    fl.next("Raw - time")
-    fl.image(
-        data.C.mean("nScans"))
-    data.reorder("t2", first=False)
-    for_plot = echo_data.C
-    for_plot.ft('t2',shift=True)
-    fl.next('FTed data')
-    fl.image(for_plot)
+data.chunk(
+    "t",
+    ["nScans", "ph_overall", "ph_diff", "tE", "t2"],
+    [
+        len(config_dict["nScans"]),
+        len(ph_overall),
+        len(ph_diff),
+        config_dict["nEchoes"],
+        nPoints,
+    ],
+)
+data.setaxis("nScans", r_[0 : len(config_dict["nScans"])])
+data.setaxis("ph_overall", ph_overall / len(ph_overall))
+data.setaxis("ph_diff", ph_diff / len(ph_diff))
```

Comment:
These are both just divided by 4

Reply:
want to discuss why in gm or later but changed